.jenkins/Jenkinsfile

```diff
@@ -135,6 +162,9 @@ pipeline {
           }
         }
         stage('Coffeelake clang-7 SGX1-FLC RelWithDebInfo') {
+          options {
+            timeout(time: 15, unit: 'MINUTES')
+          }
           agent {
```

Comment:
This is using the declarative model.  I think this timeout period starts at the start of this stage, so it will count the time that it is waiting for an executor.  Instead, you can wrap the step up in a timeout block.  Something like:
steps {
   timeout(time: 15, unit: 'MINUTES') {
      sh "..."
   }
}

By the time the pipeline has started executing the steps it has already reserved a node and an executor, so this should work.

}

Reply:
Thanks Garbiel. That's what I was thinking too. I wish this was better documented... I'll wrap the steps.